### DISCUSSION TITLE
[Feature] Prune Upload Library of External Library Duplicates

### DISCUSSION THREAD

**Discussion by undaunt** [immich-app/immich#9899] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

I, like [several others](https://github.com/immich-app/immich/discussions/4267) previously was leveraging the CLI to bulk upload content into my library from various sources, including a local copy of my iCloud photo library. There are [several](https://github.com/steilerDev/icloud-photos-sync) [tools](https://github.com/boredazfcuk/docker-icloudpd) with which to maintain a local iCloud library copy.

In light of the ability to more fully manage external library content recently (eg, editing metadata), I'm wondering if there could be a convenient, perhaps hash based way, to remove any images from the upload library if they already exist in one of the external libraries. This would allow me to keep that synced data from iCloud in a separate folder without manually having to hunt and remove thousands of photos from my current Immich library.

I am positing that a multi-external library approach will be a better wife-factor approach.

If we both use the mobile app, we will be both uploading our private and shared library content into our separate personal Immich upload libraries, doubling up disk space. However, if we only have one-pre downloaded copy of the shared library from one of those tools, and we both add it as an external library in Immich, we'll only be duplicating metadata and thumbnails, etc.

Thanks very much.

CC @agross as your duplicates finder tool could be related, or perhaps support this?

### Platform

- [X] Server
- [X] Web
- [ ] Mobile

**Comment by TheOneValen** [comment]
I do have similar issue. I use digikam to manage my photos, especially the ones from my digital camera (not phone). When I import them into the digikam library (to group them e.g. an event that I used both, camera and phone) the images would double. I have not yet found a workflow that works when using both, digikam and immich.

**Comment by xiaowuap** [comment]
Same issue，now i rolled back to use synology photos

**Comment by bo0tzz** [comment]
#4763
